### DISCUSSION TITLE
exit Gitsigns Diff This

### DISCUSSION THREAD

**Discussion by DrummyFloyd** [LazyVim/LazyVim#3067] [Q&A] [answered]
i looking for a way to exit, properly
```lua
map("n", "<leader>ghd", gs.diffthis, "Diff This")
map("n", "<leader>ghD", function() gs.diffthis("~") end, "Diff This ~")
```

i try to hit `q` like any others stuff in lazyvim but without success any idea ?

**Comment by dpetka2001** [comment, marked as answer]
This opens a vertical window to show the differences side by side so you should close the window by pressing `:q` like Neovim normally does.

**Reply by DrummyFloyd** [reply]
i did not even think about this aahah 

thank you , feel dumb about it ^^'

**Reply by nobilissimum** [reply]
> This opens a vertical window to show the differences side by side so you should close the window by pressing `:q` like Neovim normally does.

When I do `:q` from a `gitsigns.diffthis`, it puts me to a different path from before that doesn't show the current changes in the file.

Example
Original file: `foo/bar.go`
After exiting from diff this: `gitsigns:///home/myuser/codebase/.git//:0:foo/bar.go`

**Reply by alewis001** [reply]
Agreed. When the diff is opened, the split is created but your cursor remains in the original buffer. You have to switch to the gitsigns buffer and do `:q` in order to close the correct buffer (the gitsigns one instead of the one you're actually working in). 

Without thinking about it in any great detail, it feels like the active buffer should switch to the gitsigns one in this scenario to avoid this confusion. Having said that, if you start to edit the gitsigns buffer, those changes appear to be kept but they don't affect the original file so this may be a reason for the diff view to open in the original buffer, rather than the gitsigns one. I may just be displaying my lack of experience with gitsigns :)
